server/item/stack.go

```diff
@@ -115,6 +115,13 @@ func (s Stack) Damage(d int) Stack {
 	return s
 }
 
+// WithType returns a new item stack with the item type passed.
+func (s Stack) WithType(t world.Item) Stack {
```

Comment:
Not a huge fan of this method. This would allow a lot of things that should not be allowed, like creating item stacks with enchantments that aren't possible, item stacks with damage while they can't have damage, or stacks with anvil costs that can't have them.